AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift

```diff
@@ -148,6 +148,108 @@
             }
             return nil
         }
+    
+        // MARK: - Delegated Authentication
+
+        /// This method checks;
+        /// 1. if DA has been registered on the device
+        /// 2. shows an approval screen if it has been registered
+        /// else calls the completion with a failure.
+        private func performDelegatedAuthentication(_ fingerprintToken: ThreeDS2Component.FingerprintToken,
+                                                    completion: @escaping (Result<String, DelegateAuthenticationError>) -> Void) {
+            let failureHandler = { completion(.failure(DelegateAuthenticationError.authenticationFailed(cause: nil))) }
+            guard let delegatedAuthenticationInput = fingerprintToken.delegatedAuthenticationSDKInput else {
+                failureHandler()
+                return
+            }
+            
+            isDeviceRegisteredForDelegatedAuthentication(
+                delegatedAuthenticationInput: delegatedAuthenticationInput,
+                registeredHandler: { [weak self] in
+                    guard let self = self else { return }
+                    self.showApprovalScreenWhenDeviceIsRegistered(delegatedAuthenticationInput: delegatedAuthenticationInput,
+                                                                  completion: completion,
+                                                                  failureHandler: failureHandler)
+                },
+                notRegisteredHandler: failureHandler
+            )
+        }
+        
+        // MARK: Delegated Authentication Approval
+        
+        private func showApprovalScreenWhenDeviceIsRegistered(delegatedAuthenticationInput: String,
+                                                              completion: @escaping (Result<String, DelegateAuthenticationError>) -> Void,
+                                                              failureHandler: @escaping () -> Void) {
```

Comment:
also this closure should take am optional `cause` error as  parameter

Reply:
Done. Thanks ✅